### ISSUE TITLE
Fade and stop song on button press

### ISSUE THREAD

**Issue by sfxhewitt** [spotify/web-playback-sdk#57]
Is it possible for me to mod Spotify to fade out a song and stop after pressing a designated key? I'm a dj/karaoke man and would be really handy to have this feature because I hate stopping music dead and would rather have it fade out, any input is appreciated.

**Comment by fafcrumb** [comment]
Yes that would be possible with this SDK. You could implement your "fade out" by having the designated key start incrementally lowering the volume over a certain period of time with the `setVolume` method, and after the volume reaches 0, call `pause`.
However, I'd have to recommend against using this SDK if that's your only purpose. I can guarantee it'll be more trouble than it's worth. There's probably more proven solutions to this problem in that space. Maybe some program that adjusts the application volume of desktop Spotify.

**Comment by sfxhewitt** [comment]
Any idea of said program?

**Comment by fafcrumb** [comment]
Couldn't find anything that specifically does that after a quick search. The tools exist, but implementation will differ depending on your OS. On Mac OS it looks like you can use applescript to adjust volume and pause/play Spotify directly. On any OS you could also make a script that just lowers your system volume to 0 and then runs the media key pause macro (this is what media keys on your keyboard do if you have them).

**Comment by fafcrumb** [comment]
Hi @sfxhewitt could you please close this issue if you don't have any further questions?
